src/Function/Metric/Rational/CauchySequence.agda

```diff
@@ -0,0 +1,83 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Cauchy sequences on a metric over the rationals
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --cubical-compatible --guardedness #-}
+
+open import Function.Metric.Rational.Bundles
+
+module Function.Metric.Rational.CauchySequence {a ℓ} (M : Metric a ℓ) where
+
+open Metric M hiding (_≈_)
+
+open import Codata.Guarded.Stream
+open import Codata.Guarded.Stream.Properties
+open import Data.Nat.Base as ℕ using (ℕ; z≤n; s≤s)
+import Data.Nat.Properties as ℕ
+open import Data.Integer.Base using (+<+)
+open import Data.Product.Base
+open import Data.Rational.Base
+open import Data.Rational.Properties
+open import Function.Base
+open import Level
+open import Relation.Binary
+open import Relation.Binary.PropositionalEquality using (cong₂)
+
+record CauchySequence : Set a where
+  field
+    sequence : Stream Carrier
+    isCauchy : ∀ ε → {{Positive ε}} → Σ[ N ∈ ℕ ] ∀ {m n} → m ℕ.≥ N → n ℕ.≥ N → d (lookup sequence m) (lookup sequence n) < ε
```

Comment:
Should the positive proof be irrelevant? If not, it would be good to document why not?

Reply:
Yes, it should.